### PR TITLE
Tooltip cleanup

### PR DESCRIPTION
**Pull request by joshlory**
I got tripped up on the assignment inside the while-loop in a11a39135ea6f60fbddb9a689d891cf46f1128ba.  Rounding off this sharp edge.

### CODE REVIEW

File: apps/src/blockTooltips/DropletFunctionTooltip.js [hunk 1]
```diff
@@ -67,14 +68,15 @@ var DropletFunctionTooltip = function (appMsg, definition) {
   /** @type {Array.<parameterInfo>} */
   this.parameterInfos = [];
 
-  var paramId = 0;
-  var paramName;
-  while (!!(paramName = appMsg[this.parameterNameKey(paramId)] ||
-                          msg[this.parameterNameKey(paramId)])) {
+  for (var paramId = 0; ; paramId++) {
```
Comment (joshlory) [inline apps/src/blockTooltips/DropletFunctionTooltip.js]: Not super happy with the for-loop without a condition.  Open to suggestions for how to better write this.

File: apps/src/blockTooltips/DropletFunctionTooltip.js [hunk 2]
```diff
@@ -82,10 +84,13 @@ var DropletFunctionTooltip = function (appMsg, definition) {
       paramInfo.assetTooltip = definition.assetTooltip[paramId];
     }
     this.parameterInfos.push(paramInfo);
-    paramId++;
   }
 };
 
+function getLocalization(key) {
```
Comment (islemaster) [inline apps/src/blockTooltips/DropletFunctionTooltip.js]:
```javascript
/**
 * @param {string} key
 * @returns {string}
 */

File: apps/src/blockTooltips/DropletFunctionTooltip.js [hunk 3]
```diff
@@ -82,10 +84,13 @@ var DropletFunctionTooltip = function (appMsg, definition) {
...
+  return this.appMsg[key] || msg[key];
```
Comment (bcjordan) [inline apps/src/blockTooltips/DropletFunctionTooltip.js]: Should this be a prototype fn / bound to `this`?

Comment (joshlory) [inline apps/src/blockTooltips/DropletFunctionTooltip.js]: Good catch, I'm not sure how this worked at all while I was testing it...

File: apps/src/blockTooltips/DropletFunctionTooltip.js (hunk 1 above)
Comment (islemaster) [inline apps/src/blockTooltips/DropletFunctionTooltip.js]: I'm not sure it's much better, but I think semantically this is more like a while loop.
```javascript
var paramID = 0, outOfParams = false, paramName;
while (!outOfParams) {
  paramName = getLocalization(this.parameterNameKey(paramId));
  if (!paramName) {
    outOfParams = true;
  } else {
    // Create paramInfo and push into collection...
  }
  paramId++;
}
```
If you'd like to make this more readable you could isolate the 'walk until out of params' logic from the generation of the paramInfo (which could be done with `map` once you know how many you have) or somehow extract most of the inner loop to a function like `getParamInfoForIndex` that returns something falsy when you're out of params.

**Comment by islemaster** [comment]
Yay refactoring!

File: apps/src/blockTooltips/DropletFunctionTooltip.js (hunk 1 above)
Comment (breville) [inline apps/src/blockTooltips/DropletFunctionTooltip.js]: Or could you do something like this?  (not fully sure about this syntax)

for (var paramId = 0; 
	 paramName = this.getLocalization(this.parameterNameKey(paramId)); 
	 paramId++) {

**Pull request merged**